activerecord/lib/active_record/persistence.rb

```diff
@@ -460,14 +460,19 @@ def reload(options = nil)
     #   ball = Ball.new
     #   ball.touch(:updated_at)   # => raises ActiveRecordError
     #
-    def touch(*names)
+    def touch(*args)
```

Comment:
I think you should be able to use keyword arguments as @recursive-madman commented, something like this:

```ruby
def touch(*names, time: current_time_from_proper_timezone)
```

And then change the local var from `current_time` to this new `time` one, should allow you to remove the need for the unless condition. What do you think?